dts/bindings/mtd/jedec,mspi-nor.yaml

```diff
@@ -30,3 +45,18 @@ properties:
       Maximum time, in milliseconds, allowed for a single transfer on the MSPI
       bus in communication with the flash chip. The default value is the one
       that was previously hard-coded in the flash_mspi_nor driver.
+
+  use-4byte-addressing:
```

Comment:
wouldn't it make sense, to automatically enable this, when the size of the flash is to big to use 3 byte addressing.

Reply:
Buth then we could do it like in the regular spi_nor, where the address is checked if it fits in 3 bytes and if it does 3 byte addressing is used and only for addresses that don't fit 4 byte mode is used. Could then still have a option to always force 4 byte mode

Reply:
@maass-hamburg sounds like a potential improvement, perhaps not enough to block this PR at this point?